### PR TITLE
[MRG] Fix bug in n_jobs for spectral_connectivity

### PR DESCRIPTION
**Pull request by adam2392**
PR Description
--------------

Fixes: #40 

Also fixes complex number writing.

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [ ] All comments resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [ ] PR title starts with [MRG]
- [ ] [whats_new.rst](https://github.com/mne-tools/mne-connectivity/blob/main/doc/whats_new.rst) is updated
- [ ] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

**Comment by adam2392** [comment]
@drammock this should fix your issue raised.

File: mne_connectivity/base.py [hunk 1]
```diff
@@ -662,8 +662,9 @@ def save(self, fname):
 
         # save as a netCDF file
         # note this requires the netcdf4 python library
-        self.xarray.to_netcdf(fname, mode='w', format='NETCDF4',
-                              engine='netcdf4')
+        self.xarray.to_netcdf(fname, mode='w',
+                              format='NETCDF4',
+                              engine='h5netcdf')
```
Comment (larsoner) [inline mne_connectivity/base.py:667]: Is specifying the engine required? Does one work and the other not, or is one just slow, or...? A comment here will help maintainability down the road as engines may change and downstream libraries might get better or worse

**Review by larsoner — APPROVED** [review]
Otherwise LGTM

File: mne_connectivity/base.py (hunk 1 above)
Comment (adam2392) [reply mne_connectivity/base.py:667]: Specifyinng the engine is required as currently `h5netcdf` is the only engine that supports writing complex data. Xarray typically didn't include support for complex data, so their NetCDF engine doesn't work without some hacks :/

I'll add an inline comment

**Comment by drammock** [comment]
This appears to work, and fixes both the iPython print error and the n_jobs error for me.

File: doc/whats_new.rst [hunk 2]
```diff
@@ -29,12 +29,13 @@ Enhancements
 Bug
 ~~~
 
-- ...
+- Fixed bug when saving connectivity with ``n_jobs`` greater than 1 from :func:`mne_connectivity.spectral_connectivity`, by `Adam Li`_ (:gh:43)
```
Comment (drammock) [inline doc/whats_new.rst:32]:
```suggestion
- Fixed bug when saving connectivity with ``n_jobs`` greater than 1 from :func:`mne_connectivity.spectral_connectivity`, by `Adam Li`_ (:gh:`43`)
```

**Review by drammock — APPROVED** [review]

**Comment by drammock** [comment]
hah, oops, I was too slow with the official review.  Well, you'll have a rendering bug in the changelog now :)

**Comment by adam2392** [comment]
Ah oops, haha my bad. I'll fix this.

**Pull request merged**